lib/engine/game/g_18_zoo/step/buy_sell_par_shares.rb

```diff
@@ -70,6 +70,25 @@ def available_companies
 
             @game.available_companies
           end
+
+          def round_state
+            # An extra field to track whether a share was already sold from any not president since the start of the round.
+            super.merge({ share_sold: [] })
+          end
+
+          def sell_shares(entity, shares, swap: nil)
+            raise GameError, "Cannot sell shares of #{shares.corporation.name}" if !can_sell?(entity, shares) && !swap
+
+            if shares.owner == shares.corporation.owner
+              should_move = true
+            elsif @round.share_sold.include?(shares)
+              should_move = false
+            else
+              should_move = true
+              @round.share_sold << shares
+            end
```

Comment:
this if/else can be simplifies to two lines:
should_move = !@round.share_sold.include?(shares)
 @round.share_sold << shares if shares.owner != shares.corporation.owner && !@round.share_sold.include?(shares)